### ISSUE TITLE
RxAmplify.Analytics - Crash

### ISSUE THREAD

**Issue by lolucosmin** [aws-amplify/amplify-android#2943]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

RxJava

### Amplify Categories

Analytics

### Gradle script dependencies

<details>
    implementation("com.amplifyframework:core:2.24.0")
    implementation("com.amplifyframework:aws-auth-cognito:2.24.0")
    implementation("com.amplifyframework:aws-api:2.24.0")
    implementation("com.amplifyframework:aws-analytics-pinpoint:2.24.0")
    implementation("com.amplifyframework:aws-storage-s3:2.24.0")
    implementation("com.amplifyframework:aws-analytics-pinpoint:2.24.0")
    implementation("com.amplifyframework:rxbindings:2.24.0")
    coreLibraryDesugaring("com.android.tools:desugar_jdk_libs:2.0.3")
</details>

### Environment information

<details>

```
# Put output below this line
------------------------------------------------------------
Gradle 8.6
------------------------------------------------------------

Build time:   2024-02-02 16:47:16 UTC
Revision:     d55c486870a0dc6f6278f53d21381396d0741c6e

Kotlin:       1.9.20
Groovy:       3.0.17
Ant:          Apache Ant(TM) version 1.10.13 compiled on January 4 2023
JVM:          21.0.3 (JetBrains s.r.o. 21.0.3+-12282718-b509.11)
OS:           Windows 11 10.0 amd64

```

</details>

### Please include any relevant guides or documentation you're referencing

Amplify V2

### Describe the bug

Hi,  I am migrating the project to Amplify 2. I followed the steps in the documentation to initialize then to add plugins and everything was fine.

I created hubs for each plugin and for all of them I have InitializationStatus.SUCCEEDED.
I use RxAmplify in my code and was fine until I tried to record a event with  RxAmplify.Analytics.recordEvent(event).
When I run that command i get this error:

java.lang.IllegalStateException: Tried to get a plugin but that plugin was not present. Check if the plugin was added originally or perhaps was already removed.
                                                                                                    	at com.amplifyframework.core.category.Category.getPluginIfConfiguredOrThrow(Category.java:257)
                                                                                                    	at com.amplifyframework.core.category.Category.getSelectedPlugin(Category.java:252)
                                                                                                    	at com.amplifyframework.analytics.AnalyticsCategory.enable(AnalyticsCategory.java:72)
                                                                                                    	at com.bfan.sso.logic.services.analitycs.AnalyticsTracker.createEvent(AnalyticsTracker.java:33)
                                                                                                    	at com.bfan.sso.gui.base.fragment.BaseFragment.lambda$onViewCreated$0(BaseFragment.java:88)

As you can see from a base fragment I track the content name.
After a while I decide to use Amplify.Analytics.recordEvent(event) and is works fine.

Issue: RxAmplify.Analytics.recordEvent(event) has a problem.
Mention: I use everywhere in the app RxAmplify

### Code Snippet

```java
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line
java.lang.IllegalStateException: Tried to get a plugin but that plugin was not present. Check if the plugin was added originally or perhaps was already removed.
at com.amplifyframework.core.category.Category.getPluginIfConfiguredOrThrow(Category.java:257)
at com.amplifyframework.core.category.Category.getSelectedPlugin(Category.java:252)
at com.amplifyframework.analytics.AnalyticsCategory.enable(AnalyticsCategory.java:72)
at com.bfan.sso.logic.services.analitycs.AnalyticsTracker.createEvent(AnalyticsTracker.java:33)
at com.bfan.sso.gui.base.fragment.BaseFragment.lambda$onViewCreated$0(BaseFragment.java:88)

```

</details>

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

**Comment by mattcreaser** [comment]
Hi @lolucosmin, could you post your code where you configure Amplify? This exception indicates that the Analytics Plugin isn't being added, you should have something like:

```kotlin
Amplify.addPlugin(AWSCognitoAuthPlugin())
Amplify.addPlugin(AWSS3StoragePlugin())
Amplify.addPlugin(AWSPinpointAnalyticsPlugin()) // ensure this is here
Amplify.configure(...)
```

**Comment by mattcreaser** [comment]
Thanks for the additional info @lolucosmin, it seems you have uncovered a surprising bug. [This line](https://github.com/aws-amplify/amplify-android/blob/main/rxbindings/src/main/java/com/amplifyframework/rx/RxAmplify.java#L37) is a clear bug that means Analytics will _never_ work when using `RxAmplify`.

I will submit a fix for this issue and we will try to get it released as quickly as we can.

**Comment by lolucosmin** [comment]
> RxAmplify.addPlugin(...)

I use this if you check my code so still is crashing.

**Comment by mattcreaser** [comment]
Ah my mistake, you're right. It doesn't matter which you use. This will be fixed soon.

**Comment by mattcreaser** [comment]
The fix for this issue has been merged and will be included in our next release.

**Comment by mattcreaser** [comment]
This fix has been released in Amplify [2.24.1](https://github.com/aws-amplify/amplify-android/releases/tag/release_v2.24.1). Please let us know if you have any issues!
